resource/store.go

```diff
@@ -122,6 +123,10 @@ func (s *Store) Update(ctx context.Context, obj Object) (Object, error) {
 		return nil, fmt.Errorf("obj.StaticMetadata().Name must not be empty")
 	}
 
+	md := obj.CommonMetadata()
+	md.UpdateTimestamp = time.Now().UTC()
```

Comment:
While I can think of uses in our SLO app, where we might have (mis)used the capability to set this timestamp, I like this approach better where the SDK or the infra behind it takes ownership of these timestamps.

FWIW, we're interested in measuring end to end latency, and so we might have (mis)used the ability to set the "update" timestamp to when our user's initiated the update (i.e. client-side timestamp, maybe from the javascript browser code). This would feel dangerous, though, clock skew from different machines, user able to put in obviously invalid time, etc.